app/src/main/java/com/nextcloud/talk/adapters/messages/MagicSystemMessageViewHolder.java

```diff
@@ -70,22 +70,27 @@ public void onBind(ChatMessage message) {
         mentionColor = resources.getColor(R.color.textColorMaxContrast);
 
         Drawable bubbleDrawable = DisplayUtils.getMessageSelector(normalColor,
-                                resources.getColor(R.color.transparent), pressedColor,
-                                R.drawable.shape_grouped_incoming_message);
+                                                                  resources.getColor(R.color.transparent), pressedColor,
+                                                                  R.drawable.shape_grouped_incoming_message);
         ViewCompat.setBackground(bubble, bubbleDrawable);
 
         Spannable messageString = new SpannableString(message.getText());
 
         if (message.messageParameters != null && message.messageParameters.size() > 0) {
             for (String key : message.messageParameters.keySet()) {
                 Map<String, String> individualMap = message.messageParameters.get(key);
-                if (individualMap != null &&
-                        ("user".equals(individualMap.get("type")) ||
-                                "guest".equals(individualMap.get("type")) ||
-                                "call".equals(individualMap.get("type"))
-                        )) {
-                    messageString = DisplayUtils.searchAndColor(
-                            messageString, "@" + individualMap.get("name"), mentionColor);
+
+                if (individualMap != null && individualMap.containsKey("name")) {
```

Comment:
Just for the record: type, id and name are not optional. everything else depends on the type.